File: backend/src/main/java/com/dala/grademanager/jpa/Grade.java
package com.dala.grademanager.jpa;

import lombok.*;

import javax.persistence.*;
import java.util.UUID;

@Entity (name = "grades")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Grade {
    @Id
    private String id;
    private String name;
    private double grade;
    private String subject;
    private String school;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @PrePersist
    public void prePersist() {
        if (id == null) {
            id = UUID.randomUUID().toString();
        }
    }
}

File: frontend/models/Grade.ts
import {User} from "./User";

export class Grade {
    id: string | undefined;
    name: string;
    grade: number;
    subject: string;
    school: string;
    user: User | undefined;

    constructor(
        id: string | undefined,
        name: string,
        grade: number,
        subject: string,
        school: string,
        user: User | undefined
    ) {
        this.id = id;
        this.name = name;
        this.grade = grade;
        this.subject = subject;
        this.school = school;
        this.user = user;
    }
}
